recipes/c-client/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+import os
+import stat
+
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class CclientConan(ConanFile):
+    name = "c-client"
+    description = "University of Washington IMAP toolkit"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/uw-imap/imap"
+    topics = "imap", "uw-imap", "tcp-ip"
+    license = "Apache-2.0"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+    exports_sources = "patches/*"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _is_msvc(self):
+        return self._settings_build.compiler in ("Visual Studio", "msvc")
+
+    def validate(self):
+        if self._settings_build.os == "Windows" and not self._is_msvc:
+            raise ConanInvalidConfiguration(
+                "c-client is setup to build only with MSVC on Windows"
+            )
+        # FIXME: need krb5 recipe
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(
+                "c-client depends on krb5 on MacOS and it's not packaged by "
+                "Conan yet"
+            )
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if not self._is_msvc:
+            self.requires("openssl/1.1.1m")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
```

Comment:
typically we prefer out of source builds https://github.com/conan-io/conan-center-index/blob/9f21b791bce36c330be3fc06dd4c6a8aaa5cf3a5/recipes/mfast/all/conanfile.py#L91-L93